pkg/v1/tarball/write.go

```diff
@@ -416,10 +430,10 @@ func (pw *progressWriter) Close() error {
 	return io.EOF
 }
 
-// CalculateTarFileSize calculate the size a file will take up in a tar archive,
+// CalculateSingleFileInTarSize calculate the size a file will take up in a tar archive,
 // given the input data. Provided by rounding up to nearest whole block (512)
 // and adding header 512
-func CalculateTarFileSize(in int64) (out int64) {
+func CalculateSingleFileInTarSize(in int64) (out int64) {
```

Comment:
Does this need to be exported?

Reply:
Need to? Not really, but it is such a nice convenience function that it might be useful.

You kind of wish this were built into core golang `archive/tar` package.

Reply:
I agree this is useful, but I'd like to avoid exposing anything unless it's really necessary. They might accept a commit to expose some of `archive/tar`? https://github.com/golang/go/blob/8f2db14cd35bbd674cb2988a508306de6655e425/src/archive/tar/format.go#L141-L152

Reply:
Sure, I will privatize it and rebase and update. Check soon.

Reply:
I got it. Take another look. It has been a while, I might have made an error on the rebase, so please check carefully. Another pair of eyes is always helpful.